judoscale-good_job/lib/judoscale/good_job/metrics_collector.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require "judoscale/job_metrics_collector"
+require "judoscale/job_metrics_collector/active_record_helper"
+require "judoscale/metric"
+
+module Judoscale
+  module GoodJob
+    class MetricsCollector < Judoscale::JobMetricsCollector
+      def self.adapter_config
+        Judoscale::Config.instance.good_job
+      end
+
+      def collect
+        metrics = []
+        time = Time.now.utc
+
+        all_queues = ::GoodJob::JobsFilter.new({}).queues.keys
+        self.queues |= all_queues
+
+        # TODO: silence query logs for this
+        oldest_execution_time_by_queue = ::GoodJob::Execution
+          .where(performed_at: nil)
+          .group(:queue_name)
+          .pluck(:queue_name, Arel.sql("min(coalesce(scheduled_at, created_at))"))
+          .to_h
```

Comment:
@bensheldon Great feedback. Any concerns with this query? Is there any reason to include a `finished_at: nil` condition?

Reply:
I think you should include `finished_at` because it may get much better query performance. The indexes that I think this could/should use have a constraint on `finished_at`:

https://github.com/bensheldon/good_job/blob/1a37b979c54277463807007e16576e1bbe29623f/spec/test_app/db/schema.rb#LL55

Reply:
Beyond that, this looks 👍🏻 👍🏻